src/litdata/streaming/reader.py

```diff
@@ -302,6 +301,11 @@ def read(self, index: ChunkedIndex) -> Any:
             # inform the chunk has been completely consumed
             self._prepare_thread.delete([self._last_chunk_index])
 
+        if index.chunk_index != self._last_chunk_index:
+            # Close the memory-mapped file for the last chunk index
+            if isinstance(self._item_loader, TokensLoader) and self._last_chunk_index is not None:
+                self._item_loader.close(self._last_chunk_index)
+
```

Comment:
Moved it to a separate condition to run independently of `(self._config._remote_dir or self._config._compressor)`